WordPress/Classes/ViewRelated/Gravatar/UIImageView+Gravatar.swift

```diff
@@ -56,14 +58,31 @@ extension UIImageView {
         layoutIfNeeded()
 
         let size = Int(ceil(frame.width * min(2, UIScreen.main.scale)))
-        let url = gravatar.replacing(options: .init(preferredSize: .pixels(size)))?.url
-        downloadGravatar(fullURL: url, placeholder: placeholder, animate: animate)
+        guard let url = gravatar.replacing(options: .init(preferredSize: .pixels(size)))?.url else { return }
+        downloadGravatar(fullURL: url, placeholder: placeholder, animate: animate, forceRefresh: forceRefresh)
     }
 
-    private func downloadGravatar(fullURL: URL?, placeholder: UIImage, animate: Bool) {
+    private func downloadGravatar(fullURL: URL?, placeholder: UIImage, animate: Bool, forceRefresh: Bool = false) {
         wp.prepareForReuse()
         if let fullURL {
-            wp.setImage(with: fullURL)
+            var urlToDownload = fullURL
+            if forceRefresh {
+                urlToDownload = fullURL.appendingGravatarCacheBusterParam()
+            }
+
+            wp.setImage(with: urlToDownload)
+
+            if forceRefresh {
+                // If this is a `forceRefresh`, the cache for the original URL should be updated too.
+                // Because the cache buster parameter modifies the download URL.
+                Task {
+                    await wp.controller.task?.value // Wait until setting the new image is done.
+                    if let image {
+                        ImageCache.shared.setImage(image, forKey: fullURL.absoluteString) // Update the cache for the original URL
+                    }
+                }
+            }
+
```

Comment:
I'm not so sure about these changes. @kean Can you have a look please?

Reply:
The app clears caches on `onAvatarUpdated` and they will be automatically re-populated later. I'm not sure it's worth adding complexity and making `task` getter `internal` for this scenario.

Reply:
Normally we shouldn't need this yes. But unfortunately there's a delay at the backend that causes the old avatar to return for like 25 seconds after selecting a different avatar. So the old avatar comes back on the next request that doesn't have the cache buster parameter and looks like this:

<img src="https://github.com/user-attachments/assets/cf7fe31c-1d6d-4584-bab0-8f190a55e3e4" width=250/>

I reported this issue to systems and following it. In the meantime I update the memory cache for the cache buster-less version of the URL so the correct image can be shown next time.

Reply:
Yeah, this doesn't look great, especially since it replaced the new avatar with the old one. I don't see anything particular wrong with adding this bit of code, so I'm going to leave it up to you.